extensions/amp-iframely/0.1/amp-iframely.js

```diff
@@ -0,0 +1,316 @@
+     * @private {?string}
+     */
+    this.url_ = null;
+    /** @private {?string} */
+    this.key_ = null;
+
+    /**
+     * other data- options that will be passed into iFrame src
+     * @private {?Object}
+     * */
+    this.options_ = null;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    /** Populating parameters once */
+    this.widgetId_ = this.element.getAttribute('data-id');
+    this.url_ = this.element.getAttribute('data-url');
+    this.key_ = this.element.getAttribute('data-key');
+    this.options_ = this.parseOptions_();
+    let domain = 'cdn.iframe.ly';
+    const requestedDomain = this.element.getAttribute('data-domain');
+    if (requestedDomain && this.isValidDomain_(requestedDomain)) {
+      domain = requestedDomain;
+    }
+    this.base_ = `https://${domain}/`;
+    this.parseAttributes_();
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    if (this.preconnect && this.preconnect.url_) {
+      this.preconnect.url(this.base_, opt_onLayout);
+    }
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    /**
+     * Building placeholder for responsive layout without resizing
+     * or when requested via `data-img` attribute
+     */
+    const layout = this.getLayout();
+    if (
+      this.element.hasAttribute('data-img') ||
+      (layout === 'responsive' && !this.element.hasAttribute('resizable'))
+    ) {
+      /** using Iframely placeholder image */
+      const src = addParamsToUrl(
+        this.constructSrc_('/thumbnail'),
+        this.options_
+      );
+      return createElementWithAttributes(
+        this.element.ownerDocument,
+        'amp-img',
+        {
+          'src': src,
+          'placeholder': '',
+          'layout': 'fill',
+        }
+      );
+    }
+    return null;
+  }
+
+  /** @override */
+  layoutCallback() {
+    /** attach iFrame */
+    this.iframe_ = this.element.ownerDocument.createElement('iframe');
+    setStyle(this.iframe_, 'border', '0px');
+    this.iframe_.setAttribute(
+      'sandbox',
+      'allow-scripts allow-same-origin allow-popups allow-forms allow-presentation'
+    );
+    this.src_ = addParamsToUrl(this.src_, this.options_);
+    this.iframe_.src = this.src_;
+    this.applyFillContent(this.iframe_);
+    this.element.appendChild(this.iframe_);
+
+    this.unlistener_ = listen(this.win, 'message', event => {
+      if (event.source === this.iframe_.contentWindow) {
+        this.handleEvent_(this, event);
+      }
+    });
+    return this.loadPromise(this.iframe_);
+  }
+
+  /**
+   * Handles Iframely events: widget sizing, cancel, decorate cards
+   * @param {AmpIframely} me - instance of an active component
+   * @param {window.event} event - Iframely message with a method to apply
+   * */
+  handleEvent_(me, event) {
```

Comment:
I missed this, one last thing: could we please remove this `me` argument? You can use `this` directly.